### PR TITLE
Update to omniscidb 5.5.0

### PR DESCRIPTION
**Pull request by pearu**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pearu** [comment]
@jclay @andrewseidl @xmnlab please review.

Once this PR is merged, the plan is to update #22 to enable pyomniscidb as well.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,26 +32,27 @@ requirements:
     - {{ compiler('cgo') }}
     - {{ compiler('cxx') }}
     - {{ compiler("cuda") }}  # [cuda_compiler_version not in (undefined, "None")]
-    # omniscidb 5.4 is not LLVM 11 ready
-    - clangdev <11
+    # omniscidb 5.5 is not LLVM 12 ready
```
Comment (xmnlab) [inline recipe/meta.yaml:35]: note: maybe if we have this constraint `< 12` constantly, we could change it (in the future) to a variable (maybe LLVM_UNSUPPORTED_VERSION or something similar) and we could just change the variable value when needed.

**Review by xmnlab — APPROVED** [review]
LGTM! thank you @pearu

File: recipe/meta.yaml (hunk 1 above)
Comment (pearu) [reply recipe/meta.yaml:35]: I would hope that at some point we could remove the constraint on the LLVM version altogether. Meanwhile, maintaining it within the meta.yaml is straightforward, and adding the extra boilerplate is not worth it, IMHO.

**Review by andrewseidl — APPROVED** [review]

**Pull request merged**